lib/service/connect.go

```diff
@@ -920,17 +925,31 @@ func (process *TeleportProcess) newClientThroughTunnel(authServers []utils.NetAd
 	return clt, nil
 }
 
-func (process *TeleportProcess) newClientDirect(authServers []utils.NetAddr, tlsConfig *tls.Config) (*auth.Client, error) {
+func (process *TeleportProcess) newClientDirect(authServers []utils.NetAddr, tlsConfig *tls.Config, role types.SystemRole) (*auth.Client, error) {
 	var cltParams []roundtrip.ClientParam
 	if process.Config.ClientTimeout != 0 {
 		cltParams = []roundtrip.ClientParam{auth.ClientTimeout(process.Config.ClientTimeout)}
 	}
 
+	var dialOpts []grpc.DialOption
+	if role == types.RoleProxy {
+		grpcMetrics := utils.CreateGRPCClientMetrics(process.Config.Metrics.OptionalMetrics.GRPCClientLatency, prometheus.Labels{teleport.TagClient: "teleport-proxy"})
+		err := utils.RegisterPrometheusCollectors(grpcMetrics)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		dialOpts = append(dialOpts, []grpc.DialOption{
+			grpc.WithUnaryInterceptor(mw.ChainUnaryClient(metadata.UnaryClientInterceptor, om.UnaryClientInterceptor(grpcMetrics))),
+			grpc.WithStreamInterceptor(mw.ChainStreamClient(metadata.StreamClientInterceptor, om.StreamClientInterceptor(grpcMetrics))),
```

Comment:
Can we just use [grpc.WithChainUnaryInterceptor](https://pkg.go.dev/google.golang.org/grpc#WithChainUnaryInterceptor)/[grpc.WithChainStreamInterceptor](https://pkg.go.dev/google.golang.org/grpc#WithChainStreamInterceptor) and get rid of the `mw` dependency altogether?
```suggestion
			grpc.WithChainUnaryInterceptor(metadata.UnaryClientInterceptor, om.UnaryClientInterceptor(grpcMetrics)),
			grpc.WithChainStreamInterceptor(metadata.StreamClientInterceptor, om.StreamClientInterceptor(grpcMetrics)),
```

Reply:
Wasn't aware of these in the `grpc` package. Thank you for pointing them out. Have made this change.